#Defunciones en Chile entre los años 2016 y 2021"
>Nombre del Dataset: Dataset1_defunciones_2016-2021_deis.csv

##INTRODUCCIÓN
Se realiza una exploración de datos sobre muertes en Chile entre 2016-2021.
Fuente de datos: https://deis.minsal.cl/#datosabiertos


##PREGUNTAS A RESPONDER:
*   ¿Cuál fue la primera causa de muerte en cada año en Chile?
*   ¿Cuáles fueron las cinco principales causas de muerte en Chile en 2021, cada año desde 2016, y acumuladas?
*   ¿Cuál es la causa de muerte más común por región?
*   ¿Hay alguna diferencia entre las causas de muerte de hombres y mujeres?
*   ¿Cuál es la principal causa de muerte entre los niños menores de 3 años?
*   Evolución de las muertes por enfermedad COVID (2020-2021)

In [1]:
import pandas as pd
import numpy as np

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
from google.colab import drive
import os
drive.mount('/content/gdrive') #montar la carpeta

Mounted at /content/gdrive


#Carga del dataset en formtato .csv 


In [4]:
%cd '/content/gdrive/MyDrive/ProyectoDataScience/'
df_defunciones= pd.read_csv('Dataset1_defunciones_2016-2021_deis.csv',sep=',', error_bad_lines=False, index_col=False, dtype='unicode')

/content/gdrive/.shortcut-targets-by-id/1JBjZRPlZTm-Ax9WZx4quXsWlJqPbkKqm/ProyectoDataScience


/usr/local/lib/python3.8/dist-packages/IPython/core/interactiveshell.py:3326: FutureWarning: The error_bad_lines argument has been deprecated and will be removed in a future version.


  exec(code_obj, self.user_global_ns, self.user_ns)


#Entendiendo la data
##Mostar el Dataset Preliminar (Sin transformaciones)



In [5]:
df_defunciones.head()

Unnamed: 0   Año   Fecha_def    Sexo Edad Edad_num Cod_com_res     Com_resi  \
0          0  2021  2021-06-01  Hombre    1       50       13110   La Florida   
1          1  2021  2021-05-29  Hombre    1       31       13301       Colina   
2          2  2021  2021-06-02  Hombre    1       39        6101     Rancagua   
3          3  2021  2021-06-01   Mujer    1       58       13129  San Joaquín   
4          4  2021  2021-05-27  Hombre    1       46       15101        Arica   

                      Reg_resi Causa_bas_def  ... DIAG2 CAP_DIAG2  \
0    Metropolitana de Santiago          I219  ...             NaN   
1    Metropolitana de Santiago          B208  ...   NaN       NaN   
2  Del Libertador B. O'Higgins          R960  ...   NaN       NaN   
3    Metropolitana de Santiago          C539  ...   NaN       NaN   
4        De Arica y Parinacota          I219  ...   NaN       NaN   

  GLOSA_CAPITULO_DIAG2 CODIGO_GRUPO_DIAG2 GLOSA_GRUPO_DIAG2  \
0                  NaN                NaN               NaN   
1                  NaN                NaN               NaN   
2                  NaN                NaN               NaN   
3                  NaN                NaN               NaN   
4                  NaN                NaN               NaN   

  CODIGO_CATEGORIA_DIAG2 GLOSA_CATEGORIA_DIAG2 CODIGO_SUBCATEGORIA_DIAG2  \
0                    NaN                   NaN                       NaN   
1                    NaN                   NaN                       NaN   
2                    NaN                   NaN                       NaN   
3                    NaN                   NaN                       NaN   
4                    NaN                   NaN                       NaN   

  GLOSA_SUBCATEGORIA_DIAG2           lugar_def  
0                      NaN  Hospital o Clínica  
1                      NaN  Hospital o Clínica  
2                      NaN                Otro  
3                      NaN     Casa habitación  
4                      NaN     Casa habitación  

[5 rows x 28 columns]

In [6]:
df_defunciones.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 688917 entries, 0 to 688916
Data columns (total 28 columns):
 #   Column                     Non-Null Count   Dtype 
---  ------                     --------------   ----- 
 0   Unnamed: 0                 688917 non-null  object
 1   Año                        688917 non-null  object
 2   Fecha_def                  688917 non-null  object
 3   Sexo                       688917 non-null  object
 4   Edad                       688917 non-null  object
 5   Edad_num                   688917 non-null  object
 6   Cod_com_res                688917 non-null  object
 7   Com_resi                   688917 non-null  object
 8   Reg_resi                   688917 non-null  object
 9   Causa_bas_def              688917 non-null  object
 10  CAPITULO_DIAG1             688917 non-null  object
 11  GLOSA_CAPITULO_DIAG1       688917 non-null  object
 12  CODIGO_GRUPO_DIAG1         688917 non-null  object
 13  GLOSA_GRUPO_DIAG1          688917 non-null  

In [7]:
df_defunciones.describe(include='all')

Unnamed: 0     Año   Fecha_def    Sexo    Edad Edad_num Cod_com_res  \
count      688917  688917      688917  688917  688917   688917      688917   
unique     688917       6        2190       3       5      122         345   
top             0    2021  2020-06-14  Hombre       1       86       13201   
freq            1  136208         588  364323  680782    19053       17030   

           Com_resi                   Reg_resi Causa_bas_def  ...   DIAG2  \
count        688917                     688917        688917  ...  189057   
unique          345                         17          3037  ...     733   
top     Puente Alto  Metropolitana de Santiago          U071  ...           
freq          17030                     264040         39140  ...  142474   

       CAP_DIAG2                           GLOSA_CAPITULO_DIAG2  \
count      46583                                          46583   
unique         1                                              1   
top      V01-Y98  Causas externas de morbilidad y de mortalidad   
freq       46583                                          46583   

       CODIGO_GRUPO_DIAG2                          GLOSA_GRUPO_DIAG2  \
count               46583                                      46583   
unique                 32                                         32   
top               X60-X84  Lesiones autoinfligidas intencionalmente    
freq                10549                                      10549   

       CODIGO_CATEGORIA_DIAG2  \
count                   46583   
unique                    231   
top                       X70   
freq                     9414   

                                    GLOSA_CATEGORIA_DIAG2  \
count                                               46583   
unique                                                231   
top     Lesión autoinfligida intencionalmente por ahor...   
freq                                                 9414   

       CODIGO_SUBCATEGORIA_DIAG2  \
count                      46583   
unique                       732   
top                         X599   
freq                        4488   

                                 GLOSA_SUBCATEGORIA_DIAG2        lugar_def  
count                                               46583           688917  
unique                                                730                3  
top     EXPOSICIÓN A FACTORES NO ESPECIFICADOS QUE CAU...  Casa habitación  
freq                                                 4488           324940  

[4 rows x 28 columns]

In [8]:
df_defunciones.shape

(688917, 28)

Este dataset incluye un total de 688,917 registros de muerte en Chile entre 2016 y 2021.

#Analisis resumido de las variables (Valores distintos)

In [9]:
for variable in df_defunciones.columns:
    print(variable)
    print(df_defunciones[variable].unique(),"\n")

Unnamed: 0
['0' '1' '2' ... '688914' '688915' '688916'] 

Año
['2021' '2020' '2019' '2018' '2017' '2016'] 

Fecha_def
['2021-06-01' '2021-05-29' '2021-06-02' ... '2016-12-27' '2016-12-29'
 '2016-12-30'] 

Sexo
['Hombre' 'Mujer' 'Indeterminado'] 

Edad
['1' '4' '2' '3' '9'] 

Edad_num
['50' '31' '39' '58' '46' '36' '63' '41' '73' '84' '59' '82' '90' '38'
 '32' '40' '57' '85' '56' '69' '25' '78' '92' '22' '28' '75' '88' '72'
 '29' '66' '83' '87' '93' '101' '37' '79' '70' '34' '18' '1' '60' '74'
 '23' '47' '48' '99' '53' '67' '76' '54' '45' '35' '81' '42' '65' '51'
 '19' '3' '21' '27' '6' '26' '55' '64' '77' '80' '62' '20' '43' '30' '94'
 '49' '100' '44' '61' '71' '68' '33' '17' '9' '11' '91' '97' '86' '16'
 '10' '52' '2' '95' '89' '96' '111' '116' '110' '106' '107' '105' '102'
 '104' '98' '123' '103' '108' '109' '114' '24' '15' '14' '13' '12' '8' '7'
 '5' '4' '117' '112' '115' '118' '126' '121' '113' '999'] 

Cod_com_res
['13110' '13301' '6101' '13129' '15101' '2201' '7101' '13402' '1310

##Distribución de las variables en el dataset

In [10]:
for col in df_defunciones:
    print('* * * * * * * * * *',col,'* * * * * * * * * *')
    print(df_defunciones[col].value_counts(ascending=True))
    print(df_defunciones[col].value_counts(normalize=True))
    print('******************************************')

* * * * * * * * * * Unnamed: 0 * * * * * * * * * *
0         1
459273    1
459274    1
459275    1
459276    1
         ..
229640    1
229641    1
229642    1
229634    1
688916    1
Name: Unnamed: 0, Length: 688917, dtype: int64
0         0.000001
459282    0.000001
459274    0.000001
459275    0.000001
459276    0.000001
            ...   
229640    0.000001
229641    0.000001
229642    0.000001
229643    0.000001
688916    0.000001
Name: Unnamed: 0, Length: 688917, dtype: float64
******************************************
* * * * * * * * * * Año * * * * * * * * * *
2016    104026
2017    106388
2018    106796
2019    109658
2020    125841
2021    136208
Name: Año, dtype: int64
2021    0.197713
2020    0.182665
2019    0.159174
2018    0.155020
2017    0.154428
2016    0.150999
Name: Año, dtype: float64
******************************************
* * * * * * * * * * Fecha_def * * * * * * * * * *
2021-12-29     59
2017-12-29    152
2016-12-30    163
2017-12-30    182
2016-03-12    211


##Contar valores NA por columna

In [11]:
print(len(df_defunciones))

688917


In [12]:
print(df_defunciones.isna().sum())

Unnamed: 0                        0
Año                               0
Fecha_def                         0
Sexo                              0
Edad                              0
Edad_num                          0
Cod_com_res                       0
Com_resi                          0
Reg_resi                          0
Causa_bas_def                     0
CAPITULO_DIAG1                    0
GLOSA_CAPITULO_DIAG1              0
CODIGO_GRUPO_DIAG1                0
GLOSA_GRUPO_DIAG1                 0
CODIGO_CATEGORIA_DIAG1            0
GLOSA_CATEGORIA_DIAG1             0
CODIGO_SUBCATEGORIA_DIAG1         0
GLOSA_SUBCATEGORIA_DIAG1          0
DIAG2                        499860
CAP_DIAG2                    642334
GLOSA_CAPITULO_DIAG2         642334
CODIGO_GRUPO_DIAG2           642334
GLOSA_GRUPO_DIAG2            642334
CODIGO_CATEGORIA_DIAG2       642334
GLOSA_CATEGORIA_DIAG2        642334
CODIGO_SUBCATEGORIA_DIAG2    642334
GLOSA_SUBCATEGORIA_DIAG2     642334
lugar_def                   

##Eliminar columnas vacias

In [13]:
df_defunciones = df_defunciones.drop(columns=["CAP_DIAG2",
"GLOSA_CAPITULO_DIAG2",
"CODIGO_GRUPO_DIAG2",
"CODIGO_CATEGORIA_DIAG2",
'GLOSA_CATEGORIA_DIAG2',
"CODIGO_SUBCATEGORIA_DIAG2",
"GLOSA_SUBCATEGORIA_DIAG2"])
df_defunciones.shape   # Se eliminan 7 columnas vacias

(688917, 21)

# Notas
> 1.- la variable "Causa básica de muerte" ("Causa_bas_def"). Este valor está en codificación CIE10. El equivalente español de la codificación inglesa ICD-10. Los cuales estan descritos en el siguiente enlace https://icd10coded.com/search/